lib/jsonschema.lua

```diff
@@ -895,34 +895,23 @@ generate_validator = function(ctx, schema)
   if schema.minimum or schema.maximum or schema.multipleOf or schema.exclusiveMinimum or schema.exclusiveMaximum then
     ctx:stmt(sformat('if %s == "number" then', datatype))
 
-    if schema.minimum then
-      local op = '<'
-      local msg = 'greater'
-      ctx:stmt(sformat('  if %s %s %s then', ctx:param(1), op, schema.minimum))
-      ctx:stmt(sformat('    return false, %s("expected %%s to be %s than %s", %s)',
-                       ctx:libfunc('string.format'), msg, schema.minimum, ctx:param(1)))
+    local addRangeCheck = function (op, reference, msg)
```

Comment:
This way of writing has a bad performance. You can use this way:

```lua
local function(ctx, op, reference, msg)
   ... ...
end

generate_validator = function(ctx, schema)
    ...
    if schema.minimum then
      addRangeCheck(ctx, '<', schema.minimum, 'at least')
    end
    ...
end

Reply:
Sorry for delay. I incorporated your improvement.